### ISSUE TITLE
(depends-on "foo.cert") doesn't work well

### ISSUE THREAD

**Issue by jaredcdavis** [acl2/acl2#383]
I think I have a working demo of a `defpm` build oddity.  From a completely clean ACL2 checkout, after building ACL2, `cd books; cert.pl misc/defpm` seems to fail to build `misc/arithmetic-top-theory.lisp` with the following message:

```
   | ACL2 Error in ( INCLUDE-BOOK "arithmetic/top" ...):  There is no certificate
   | on file for "/home/jared/xb/books/arithmetic/top.lisp".  This is illegal
   | because we are currently attempting certify-book; see :DOC certify-
   | book.
   | 
   | 
   | Summary
   | Form:  ( INCLUDE-BOOK "arithmetic/top" ...)
   | Rules: NIL
   | Warnings:  Compiled file
   | Time:  0.00 seconds (prove: 0.00, print: 0.00, other: 0.00)
```

I believe the cause of this is the very unusual line in defpm:

```
; (depends-on "arithmetic-top-theory.cert")
```

In the resulting `Makefile-tmp`, I'm seeing:

```
misc/defpm.cert : \
     xdoc/top.cert \
     misc/defpm.lisp \
     misc/arithmetic-top-theory.cert
```

But there doesn't seem to be any dependency information for `misc/arithmetic-top-theory.cert`!  Because of this, `make` tries to build it before certifying `arithmetic/top`.

I'm not sure whether or not this *should* work, but a simple workaround is to replace the `depends-on` line with a phony `include-book`, e.g.,

```
#||
(include-book "arithmetic-top-theory")
||#
```

I plan to make this change, which should solve the immediate problem.  In the long run, perhaps `cert.pl` should either:

1. Simply reject any `depends-on` lines to target `.cert` files and say to use a commented `include-book` directive instead, or
2. Make sure that any such `depends-on` lines are getting processed like `include-book` directives.

**Comment by jaredcdavis** [comment]
Workaround for `defpm` in f384a69114b32411bb731bbbe97b667b30e8231f

**Comment by ragerdl** [comment]
Ah yes, this was the problem I've seen.

**Comment by solswords** [comment]
Yes, depends-on is supposed to be used for non-book dependencies (cert.pl doesn't automatically scan for the dependencies of the target).  Jared's workaround is what I'd do -- include-book inside a multiline comment.  I'll add a thing in the cert.pl documentation under "depends-on" to clarify.

**Comment by jaredcdavis** [comment]
I think this is fixed as of 5fc3078d00816dcfd77de07b3ac5b9572fc520cc, which adds documentation and adjusts `certlib.pl` to cause an error if someone tries to do this again.

**Comment by ragerdl** [comment]
Thank you for causing an error, as it will save me the trouble of reading
this github issue.
